---
title: "Inflammatory Response from Cancer Immunotherapy"
author: "Scott Ginebaugh"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE, fig.width = 7, fig.height = 7, cache = FALSE)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
```

```{r load_libraries, include = FALSE, message = FALSE, warning = FALSE}
library(Seurat)
library(qs)
library(magrittr)
library(dplyr)
library(harmony)
library(stringr)
library(tidyverse)
library(tibble)
set.seed(42555)

library(scRepertoire)
library(ggraph)

library(ggplot2)
library(ggpmisc)
library(ggprism)
library(ggrepel)
library(knitr)

library(clusterProfiler)
library(msigdbr)

source("code/utils/TCR_utils.R")
```

```{r functions, include = FALSE, message = FALSE, warning = FALSE}
prep_data <- function(seurat_obj){
  seurat_obj <- add_contig_barcodes(seurat_obj)
  seurat_obj$patient_group <- factor(seurat_obj$patient_group,
                                     levels = c("Control","CPI no colitis","CPI colitis"),
                                     ordered = FALSE)
  donor_ids <- c(paste0("CT",1:8),
                 paste0("NC",1:6),
                 paste0("C",1:8))
  donor_ids <- donor_ids[donor_ids %in% seurat_obj$patient]
  
  seurat_obj$donor_id <- factor(seurat_obj$donor_id, 
                                levels = donor_ids, 
                                ordered = FALSE)
  return(seurat_obj)
}

get_donor_colors <- function(meta_data){
    axis_colors <- sapply(levels(meta_data$donor_id), function(x) {
      if (grepl("^CT", x)) return(ctr_col)     
      else if (grepl("^C[0-9]", x)) return(cpi_col)  
      else if (grepl("^NC", x)) return(cpi_nc_col)    
    })
    return(axis_colors)
}

colitis_pseudobulk <- function(seurat_obj){
  pb_dat <- AggregateExpression(seurat_obj, assay = "RNA", return.seurat = TRUE, group.by = "donor_id")
  pb_dat$patient_group <- sapply(pb_dat$donor_id, function(x) {
        if (grepl("^CT", x)) return("Control")     
        else if (grepl("^C[0-9]", x)) return("CPI colitis")  
        else if (grepl("^NC", x)) return("CPI no colitis")    
      })
  Idents(pb_dat) <- pb_dat$patient_group
  colitis_deg <- FindMarkers(pb_dat, 
                            ident.1 = "CPI colitis", 
                            ident.2 = c("CPI no colitis", "Control"), 
                            test.use = "DESeq2")
  colitis_deg$gene_name <- rownames(colitis_deg)
  return(colitis_deg)
}

add_summary_metadata <- function(metadata){
  metadata$sample <- word(metadata$patient_celltype,1,1,"_")
  metadata$celltype <- word(metadata$patient_celltype,2,-1,"_")
  metadata$patient_group <- sapply(metadata$sample, function(x) {
      if (grepl("^CT", x)) return("Control")     
        else if (grepl("^C[0-9]", x)) return("CPI colitis")  
        else if (grepl("^NC", x)) return("CPI no colitis")    
      })
  return(metadata)
}

get_patient_group_factor <- function(patient_group){
  patient_group <- factor(patient_group, levels = c("Control","CPI no colitis","CPI colitis"), ordered = FALSE)
  return(patient_group)
}
```

```{r import_data, include = FALSE, message = FALSE, warning = FALSE}
sc_all <- qread("data/processed/annotated_seurat.qs")
seurat <- qread("data/processed/annotated_Tcell.qs")
cd4 <- qread("data/processed/annotated_cd4.qs")
cd8 <- qread("data/processed/annotated_cd8.qs")

sc_all <- prep_data(sc_all)
seurat <- prep_data(seurat)
cd4 <- prep_data(cd4)
cd8 <- prep_data(cd8)

combined_tcr <- read.csv("data/processed/combined_tcr.csv")

sc_meta <- seurat@meta.data
sc_meta <- sc_meta[,c("barcode_contig","patient_group","treatment","age","sex","malignancy",
                      "recist_response","overal_surival_months_from_cpi_initiation",
                      "diarrhea_grade_CTCAE","mayo_endoscopic_score_MES",
                      "time_from_first_treatment_to_symptom_onset_days","infliximab_required_for_treatment",
                      "azimuth_celltype_1","azimuth_celltype_2","azimuth_celltype_3",
                      "annotation_level1","annotation_level2","annotation_level3")]
colnames(sc_meta)[1] <- "barcode"

combined_tcr_merge <- left_join(combined_tcr, sc_meta, by = "barcode")

split_tcr <- split(combined_tcr, f = combined_tcr_merge$donor_id)

# set colors for plotting
ctr_col = "#00bfa0"
cpi_nc_col = "#ffa300"
cpi_col = "#9b19f5"

colors <- c("Control" = ctr_col, 
            "CPI no colitis" = cpi_nc_col, 
            "CPI colitis" = cpi_col) 
```

# Introduction

A growing field of interest in immunology and is the study of T cell subtype heterogeneity and T-cell receptor (TCR) diversity. This is of particular interest in immuno-oncology, where a deep understanding of T-cell and TCR diversity is essential to understanding the tumor microenvironment and, importantly, for developing [new cancer therapies](https://jhoonline.biomedcentral.com/articles/10.1186/s13045-021-01115-0).

Recent advances in 5' single-cell sequencing, such as 10X Genomics [Chromium Immune Profiling](https://www.10xgenomics.com/products/single-cell-immune-profiling), enables simultaneous profiling of gene expression and TCR [V(D)J](https://en.wikipedia.org/wiki/V(D)J_recombination) sequences.

Here, we reanalyze a mutliomic scRNA+scTCR dataset from [Luoma et al.](https://www.cell.com/cell/fulltext/S0092-8674(20)30688-7?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867420306887%3Fshowall%3Dtrue) In this manuscript, they investigate immune cell behavior in the colon during CTLA-4 and PD-1/PD-L1 checkpoint inhibitor (CPI) treatment. This is of particular importance because 60% of patients treated with these checkpoint blockades experience severe treatment-limiting toxicities -- particularly in the gastrointestinal mucosa. 10-20% of patients treated with CTLA-4 checkpoint inhibitors experience severe, and potentially life-threatening, colon inflammation (a.k.a. colitis).

# Results

## Dataset Overview

This dataset contains colon biopsies from n = 8 melanoma patients with colitis, n = 6 melanoma patients without colitis, and n = 8 healthy adults who underwent screening colonoscopies. Of the 6 melanoma patients without colitis, 3 had enteritis (inflammation of the small intestine) and 3 did not.

Following biopsy, the colon tissue was enzymatically digested, FACS sorted into a CD3+ T-cell population, and then sequenced with 10X Genomics 5' single-cell immune profiling. For 16 of the patients, a CD45+ mononuclear cell population was also sorted and sequenced.

```{r allcell_umap_celltype_1, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 6, fig.height = 5}
DimPlot(sc_all, group.by = c("annotation_level1"), label = TRUE, raster = TRUE) + ggtitle("Cell Type")
DimPlot(sc_all, group.by = c("patient_group"), label = TRUE, raster = TRUE, cols = colors) + ggtitle("Patient Group")

```

```{r allcell_umap_celltype_2, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
DimPlot(sc_all, group.by = c("flow_cell", "sex"), label = FALSE, raster = TRUE)
```


When looking only at the CD45+ separated cells, we see that colitis patients have a higher proportion of T-cells than non-colitis patients.

```{r cd45_cell_prop, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
meta_45 <- sc_all@meta.data[sc_all@meta.data$flow_cell == "CD45", ]
meta_45$donor_id <- factor(meta_45$donor_id)

axis_colors <- get_donor_colors(meta_45)


ggplot(meta_45, aes(x = donor_id, fill = annotation_level1)) +
  geom_bar(position = "fill") +
  theme_prism() +
  theme(axis.text.x = element_text(color = axis_colors, angle = 90, vjust = 0.5, hjust = 1)) +
  ylab("Cell Proportion in CD45+ Separated Cells") +
  xlab("Patient ID")

```

Additionally, we can see that there are large differences between treatment groups within different celltypes.

```{r cluster_umap, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
DimPlot(sc_all[,sc_all@meta.data$flow_cell == "CD45"], group.by = "RNA_snn_res.0.4", split.by = "patient_group") +
  ggtitle("")

```

## T-cell subclustering

### T-cell overview

Next, we investigate the T-cells in greater detail. We see that the T-cells generally separate into CD8 and CD4 subtypes. However, it isn't perfect -- there are some clusters (e.g. cluster 10) which contain both CD4 & CD8 cells, and there are some clusters which contain neither CD4 or CD8 cells (e.g. cluster 18), or contain only very low expression of a marker gene (e.g. cluster 13). 

```{r T_cell_both_umap, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
DimPlot(seurat, group.by = c("patient_group", "flow_cell","sex"), label = FALSE, raster = TRUE, cols = colors)
DimPlot(seurat, group.by = c("RNA_snn_res.1"), label = TRUE, raster = TRUE)
FeaturePlot(seurat, min.cutoff = "q1", max.cutoff = "q99", features = c("CD4","CD8A","CD8B"))
FeaturePlot(seurat, min.cutoff = "q1", max.cutoff = "q99", features = c("CD3E","CD3D","CD3G"))
```

To remedy this, we instead use [Azimuth](https://www.cell.com/cell/fulltext/S0092-8674(21)00583-3) to annotate the cell types. We found that this worked very well for the initial labeling of CD4/CD8/other T-cell subtypes, but did a poor job of capturing heterogeneity in sub-subtypes. So, we next separated the CD4 and CD8 T-cells based on the Azimuth level 1 annotation, and studied them separately.

```{r Azimuth_umap, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
seurat$annotation_level2[is.na(seurat$annotation_level2)] <- "Other"
DimPlot(seurat, group.by = c("azimuth_celltype_1","azimuth_celltype_2","annotation_level2"), label = FALSE, raster = FALSE)
```

```{r Azimuth_cell_prop, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
axis_colors <- get_donor_colors(seurat@meta.data)
ggplot(seurat@meta.data, aes(x = donor_id, fill = annotation_level2)) +
  geom_bar(position = "fill") +
  theme_prism() +
  theme(axis.text.x = element_text(color = axis_colors, angle = 90, vjust = 0.5, hjust = 1))
```

### CD8+ T-cell subclustering

First, we analyze the CD8+ Tcells. We see distinct clusters based on known celltype markers, and we see a large difference in cell-type proportions of CPI colitis patients compared to both controls and CPI without colitis. In particular, we see a large increase in the cycling and cytotoxic effector cells in CPI colitis, and a loss of Trm IEL and Trm LP1 cells.

```{r CD8_umap, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 5, fig.height = 5}
DimPlot(cd8, group.by = c("patient_group"), label = TRUE, cols = colors) +
  ggtitle("Patient Group")
DimPlot(cd8, group.by = c("annotation_level3"), label = TRUE) +
  ggtitle("CD8+ T-cell sub-clustering")
```

```{r CD8_cell_prop, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
axis_colors <- get_donor_colors(cd8@meta.data)
ggplot(cd8@meta.data, aes(x = donor_id, fill = annotation_level3)) +
  geom_bar(position = "fill") +
  theme_prism() +
  theme(axis.text.x = element_text(color = axis_colors, angle = 90, vjust = 0.5, hjust = 1))

cd8_prop <- prop.table(table(cd8$patient_group, cd8$annotation_level3), margin = 1) %>%
  data.frame()
cd8_prop$Var1 <- factor(cd8_prop$Var1, levels = c("Control","CPI no colitis","CPI colitis"))
ggplot(cd8_prop, aes(x = Var2, y = Freq, fill = Var1)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  theme_prism() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  scale_fill_manual(values = colors) +
  ylab("Cell proportion") +
  xlab("CD8 cell subtype")
```

### CD4+ T-cell overview

Next, we analyze the CD4+ T-cells. We see distinct clusters based on known celltype markers, and we see a large difference in cell-type proportions of CPI colitis patients compared to both controls and CPI without colitis. In particular, we see a notable decrease in the proportion of Tissue Resident Memory cells (Trm) and a large increase in the proportion of Th1 effector cells, cycling cells, and regulatory T-cells

```{r CD4_umap, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 5, fig.height = 5}
DimPlot(cd4, group.by = c("patient_group"), label = TRUE, cols = colors) +
  ggtitle("Patient Group")
DimPlot(cd4, group.by = c("annotation_level3"), label = TRUE) +
  ggtitle("CD4+ T-cell sub-clustering")
```
```{r CD4_cell_prop, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
axis_colors <- get_donor_colors(cd4@meta.data)
ggplot(cd4@meta.data, aes(x = donor_id, fill = annotation_level3)) +
  geom_bar(position = "fill") +
  theme_prism() +
  theme(axis.text.x = element_text(color = axis_colors, angle = 90, vjust = 0.5, hjust = 1)) +
  ylab("Cell Proportions")

cd4_prop <- prop.table(table(cd4$patient_group, cd4$annotation_level3), margin = 1) %>%
  data.frame()
cd4_prop$Var1 <- factor(cd4_prop$Var1, levels = c("Control","CPI no colitis","CPI colitis"))
ggplot(cd4_prop, aes(x = Var2, y = Freq, fill = Var1)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  theme_prism() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  scale_fill_manual(values = colors) +
  ylab("Cell Proportions") +
  xlab("CD4 cell subtype")
```
## T-cell checkpoint gene expression

We next investigated the expression of immune checkpoint genes and inhibitory receptors in T-cell subclusters. We see that there is an increased expression of checkpoint-related genes in both CD8 and CD4 T-cells. In a pseudobulk comparison of colitis vs non-colitis + control samples, we see that CTLA4, HAVCR2, LAG3, and CD38 are significantly upregulated in colitis CD8+ T-cells, but only HAVCR2 and CD38 are upregulated in colitis CD4+ T-cells.

```{r CD8_checkpoint_expr, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
# having a metadata column named CTLA4 causes errors when trying to plot CTLA4 gene expression
cd8_plot <- cd8
cd8_plot$CTLA4 <- NULL

cd4_plot <- cd4
cd4_plot$CTLA4 <- NULL

checkpoint_genes <- c("CTLA4","PDCD1","HAVCR2","LAG3","TIGIT","CD38")

VlnPlot(cd8_plot, features = checkpoint_genes, pt.size = 0, 
        group.by = "patient_group",  cols = colors, adjust = 2)

VlnPlot(cd4_plot, features = checkpoint_genes, pt.size = 0, 
        group.by = "patient_group",  cols = colors, adjust = 2)

# VlnPlot(cd8_plot, features = checkpoint_genes, pt.size = 0, 
#         group.by = "annotation_level3")
# 
# VlnPlot(cd4_plot, features = checkpoint_genes, pt.size = 0, 
#         group.by = "annotation_level3")


VlnPlot(cd8_plot, features = checkpoint_genes, pt.size = 0, 
        group.by = "annotation_level3", split.by = "patient_group",
        stack = TRUE, flip = TRUE, cols = colors)

VlnPlot(cd4_plot, features = checkpoint_genes, pt.size = 0, 
        group.by = "annotation_level3", split.by = "patient_group",
        stack = TRUE, flip = TRUE, cols = colors)

FeaturePlot(cd8_plot, min.cutoff = "q1", max.cutoff = "q99", features = checkpoint_genes)
FeaturePlot(cd4_plot, min.cutoff = "q1", max.cutoff = "q99", features = checkpoint_genes)

```


```{r pseudobulk_prep, include = FALSE, message = FALSE, warning = FALSE}
cd8_pb <- colitis_pseudobulk(cd8)
cd4_pb <- colitis_pseudobulk(cd4)

cd8_pb[checkpoint_genes,]
cd4_pb[checkpoint_genes,]

```

In general, we see a strong correlation between colitis-associated changes in gene expression between CD8+ and CD4+ T-cells.

```{r pseudobulk_comparison, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 11}
de_merge <- inner_join(cd4_pb, cd8_pb, by = "gene_name", suffix = c(".CD4",".CD8"))
de_merge <- de_merge[!is.na(de_merge$p_val_adj.CD4) & !is.na(de_merge$p_val_adj.CD8),]

ggplot(de_merge, aes(x = avg_log2FC.CD4, y = avg_log2FC.CD8)) +
  geom_point() +
  theme_prism() +
  stat_poly_line() +
  stat_poly_eq() +
  geom_text_repel(data = subset(de_merge,
                         avg_log2FC.CD4 > 2 |
                           avg_log2FC.CD4 < 0.5 |
                           avg_log2FC.CD8 > 2 |
                           avg_log2FC.CD8 < -1.2),
                  mapping = aes(x = avg_log2FC.CD4, y = avg_log2FC.CD8, label = gene_name)) +
  xlab("CD4+ log2FC") +
  ylab("CD8+ log2FC")
```


### CD8+ TCR analysis

```{r tcr_prep1, include = FALSE, message = FALSE, warning = FALSE}
cd8_tcr <- combined_tcr_merge[combined_tcr_merge$annotation_level2 %in% "CD8_Tcell",]
cd8_split_tcr <- split(cd8_tcr, f = combined_tcr_merge$donor_id)
cd8_merge <- cd8[,cd8$flow_cell == "CD3"]
cd8_merge <- combineExpression(cd8_split_tcr,
                              cd8_merge,
                              cloneCall = "gene",
                              proportion = TRUE,
                              group.by = "patient_group",
                              filterNA = TRUE)
cd8_merge$cloneSize <- as.character(cd8_merge$cloneSize)
cd8_merge$cloneSize[cd8_merge$clonalFrequency == 1] <- "Single"
cd8_merge$expanded <- ifelse(cd8_merge$cloneSize == "Single", "Single","Expanded")
Idents(cd8_merge) <- cd8_merge$annotation_level3

cd4_tcr <- combined_tcr_merge[combined_tcr_merge$annotation_level2 %in% "CD4_Tcell",]
cd4_split_tcr <- split(cd4_tcr, f = combined_tcr_merge$donor_id)
cd4_merge <- cd4[,cd4$flow_cell == "CD3"]
cd4_merge <- combineExpression(cd4_split_tcr,
                              cd4_merge,
                              cloneCall = "gene",
                              proportion = TRUE,
                              group.by = "patient_group",
                              filterNA = TRUE)
cd4_merge$cloneSize <- as.character(cd4_merge$cloneSize)
cd4_merge$cloneSize[cd4_merge$clonalFrequency == 1] <- "Single"
cd4_merge$expanded <- ifelse(cd4_merge$cloneSize == "Single", "Single","Expanded")
Idents(cd4_merge) <- cd4_merge$annotation_level3

```

```{r TCR_stats, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 5, fig.height = 5}
cd_all_meta <- rbind(cd8_merge@meta.data, cd4_merge@meta.data)
cd_all_meta$patient_celltype <- paste0(cd_all_meta$patient, "_", cd_all_meta$azimuth_celltype_1)
cd_all_meta$patient_subcelltype <- paste0(cd_all_meta$patient, "_", cd_all_meta$annotation_level3)

unique_clones <- cd_all_meta %>%
  group_by(patient_celltype) %>%
  summarize(unique_clones = n_distinct(CTaa))
unique_clones <- add_summary_metadata(unique_clones)
unique_clones$patient_group <- get_patient_group_factor(unique_clones$patient_group)


expansion_ratio <- cd_all_meta %>%
  group_by(patient_celltype) %>%
  summarize(
    expanded_unique = n_distinct(CTaa[clonalFrequency > 1]),
    single_count = sum(clonalFrequency == 1),
    ratio = expanded_unique / (single_count + expanded_unique)
  )
expansion_ratio <- add_summary_metadata(expansion_ratio)
expansion_ratio$patient_group <- get_patient_group_factor(expansion_ratio$patient_group)

cd8_expanded <- cd_all_meta %>%
  dplyr::filter(azimuth_celltype_1 == "CD8 T") %>%
  group_by(patient_subcelltype) %>%
  summarize(
    expanded_unique = n_distinct(CTaa[clonalFrequency > 1]),
    single_count = sum(clonalFrequency == 1),
    ratio = expanded_unique / (single_count + expanded_unique)
  )
colnames(cd8_expanded)[1] <- "patient_celltype"
cd8_expanded <- add_summary_metadata(cd8_expanded)
cd8_expanded$patient_group <- get_patient_group_factor(cd8_expanded$patient_group)

cd4_expanded <- cd_all_meta %>%
  dplyr::filter(azimuth_celltype_1 == "CD4 T") %>%
  group_by(patient_subcelltype) %>%
  summarize(
    expanded_unique = n_distinct(CTaa[clonalFrequency > 1]),
    single_count = sum(clonalFrequency == 1),
    ratio = expanded_unique / (single_count + expanded_unique)
  )
colnames(cd4_expanded)[1] <- "patient_celltype"
cd4_expanded <- add_summary_metadata(cd4_expanded)
cd4_expanded$patient_group <- get_patient_group_factor(cd4_expanded$patient_group)


ggplot(unique_clones, aes(x = patient_group , y = unique_clones, fill = celltype  )) +
  geom_boxplot() +
  theme_prism() +
  theme(axis.text.x = element_text(color = colors, angle = 90, vjust = 0.5, hjust = 1)) +
  ylab("# unique clonotypes per patient") +
  xlab("")

ggplot(expansion_ratio, aes(x = patient_group , y = expanded_unique, fill = celltype  )) +
  geom_boxplot() +
  theme_prism() +
  theme(axis.text.x = element_text(color = colors, angle = 90, vjust = 0.5, hjust = 1)) +
  ylab("# unique expanded clonotypes\nper patient") +
  xlab("")

ggplot(expansion_ratio, aes(x = patient_group , y = ratio, fill = celltype)) +
  geom_boxplot() +
  theme_prism() +
  theme(axis.text.x = element_text(color = colors, angle = 90, vjust = 0.5, hjust = 1)) +
  ylab("Expanded clonotypes\n(% of total) per patient") +
  xlab("")


ggplot(cd8_expanded, aes(x = celltype, y = expanded_unique, fill = patient_group)) +
  geom_boxplot() +
  theme_prism() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  ylab("# unique expanded clonotypes\nper patient per celltype") +
  xlab("") +
  scale_fill_manual(values = colors)

ggplot(cd4_expanded, aes(x = celltype, y = expanded_unique, fill = patient_group)) +
  geom_boxplot() +
  theme_prism() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  ylab("# unique expanded clonotypes\nper patient per celltype") +
  xlab("") +
  scale_fill_manual(values = colors)


```

```{r CD8_tcr_1, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
DimPlot(cd8_merge, group.by = "cloneSize")
clonalOverlay(cd8_merge, reduction = "umap", cutpoint = 1, bins = 10, facet.by = "patient_group")
clonalNetwork(cd8_merge, reduction = "umap", group.by = "annotation_level3",
              filter.clones = 2000, filter.identity = "Cytotoxic_effector", cloneCall = "aa", filter.graph = TRUE)
library(circlize)
library(scales)
circles <- getCirclize(cd8_merge, group.by = "annotation_level3")
grid.cols <- hue_pal()(length(unique(cd8_merge$annotation_level3)))
names(grid.cols) <- unique(cd8_merge$annotation_level3)

#Graphing the chord diagram
chordDiagram(circles, self.link = 1, grid.col = grid.cols)
chordDiagram(circles, self.link = 1, grid.col = grid.cols, directional = 1, direction.type = "arrows",
    link.arr.type = "big.arrow")
```

```{r CD8_tcr_2, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 5}
clonalDiversity(cd8_split_tcr, cloneCall = "aa", group.by = "patient_group")
clonalOverlap(cd8_split_tcr, cloneCall = "aa", group.by = "annotation_level3", method = "raw") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

```{r top_aa, echo = FALSE, message = FALSE, warning = FALSE, fig.width = 11, fig.height = 11}
top_aa <- cd8_merge@meta.data %>%
  dplyr::filter(annotation_level3 == "Cytotoxic_effector") %>%
  pull(CTaa) %>%
  table()
top_aa <- top_aa[order(top_aa, decreasing = TRUE)]

cd8_merge$top_aa <- NA
cd8_merge$top_aa[cd8_merge$CTaa %in% names(top_aa)[1:10]] <- cd8_merge$CTaa[cd8_merge$CTaa %in% names(top_aa)[1:10]]

cells_list <- list()
for (i in 1:20){
  cells_list[[i]] <- which(cd8_merge$CTaa == names(top_aa)[i])
}
names(cells_list) <- names(top_aa)[1:20]


DimPlot(cd8_merge, group.by = "top_aa", raster = FALSE, pt.size = 4, na.value = "black")
DimPlot(cd8_merge, cells.highlight = cells_list, cols.highlight = hue_pal()(length(cells_list)),
        raster = FALSE, sizes.highlight = 3) + 
theme(
    legend.position = "bottom",
    legend.text = element_text(size = 9),  
    legend.key.size = unit(0.3, "cm"),     
    legend.spacing.x = unit(0.2, "cm"),    
    legend.margin = margin(0, 0, 0, 0),    
    legend.box = "horizontal",             
    legend.text.align = 0                  
  ) +
  guides(color = guide_legend(nrow = 7))  

```


















